src/main/java/org/tarantool/TarantoolClientImpl.java

```diff
@@ -174,8 +173,11 @@ public void run() {
                         readThread();
                     } finally {
                         state.release(StateHelper.READING);
-                        if (state.compareAndSet(0, StateHelper.RECONNECT))
-                            LockSupport.unpark(connector);
+                        // avoid a case when this thread falls asleep here
+                        // after READING flag released and then can pollute the state
```

Comment:
To be precise: I guess, 'pollute the state' means to set RECONNECT again?

Reply:
I slightly renamed the vars to be more self-documented.